File: club_crm/club_crm/report/membership_report/membership_report.js
// Copyright (c) 2016, Blue Lynx and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Membership Report"] = {
	"filters": [
		{
            "fieldname": "membership_category",
            "fieldtype": "Link",
            "label": "Membership Category",
            "options": "Memberships Category"
        },
		{
            "fieldname": "memberships_type",
            "fieldtype": "Select",
            "label": "Membership Type",
            "options": "\nNew\nEarly Renew\nLate Renew"
        },
		{
            "fieldname": "membership_type",
            "fieldtype": "Link",
            "label": "Membership Type",
            "options": "Memberships Type"
        },
		{
            "fieldname": "membership_plan",
            "fieldtype": "Link",
            "label": "Membership Plan",
            "options": "Memberships Plan"
        },
	]
};

File: club_crm/club_crm/report/membership_report/membership_report.py
from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
	columns, query = [], []
	columns = get_columns(filters)
	query = get_data(filters)
	return columns, query

def get_columns(filters):
	columns = [
		_("Primary Client ID") + ":Data:250",
		_("Primary Client Name") + "::80",
		_("Member ID") + "::90",
		_("Membership Plan") + ":Data:110",
		_("Memberships Type") + ":Data:100",
		_("Membership Category") + ":Data:100",
		_("Membership Type") +":Data:100",
	]

	# if filters.get('membership_type') == "Family Membership":
	# 	columns.insert(8, _("Client ID") + ":Data:220"),
	# 	columns.insert(9, _("Client  Name") + ":Data:100")

	return columns

def get_data(filters):
	conditions = get_conditions(filters)
	query = frappe.db.sql("""select 
								mem.primary_client_id,
								mem.primary_client_name,
								mem.membership_id,
								mem.membership_plan,
								mem.memberships_type,
								mem.membership_category,
								mem.membership_type
							FROM `tabMemberships` mem
							where mem.membership_status != 'Cancelled' %s """ % 
							conditions, filters, as_dict=1)
							
	return query

def get_conditions(filters):
	conditions = ""
	if filters.get("membership_type"): conditions += " and mem.membership_type = %(membership_type)s"
	if filters.get('memberships_type'): conditions += "and mem.memberships_type = %(memberships_type)s"
	return conditions

